code/go/0chain.net/smartcontract/stakepool/stakepool.go

```diff
@@ -844,7 +844,7 @@ func validateLockRequest(t *transaction.Transaction, sp AbstractStakePool, vs Va
 		}
 	}
 
-	cstate.WithActivation(balances, "hard_fork_1", beforeFunc, afterFunc)
+	err = cstate.WithActivation(balances, "hard_fork_1", beforeFunc, afterFunc)
```

Comment:
looks like this will shadow the `err` from `before` and `after` function, perhaps we will need to return error for `before` and `after` function. In that way, if any error happens in the before/after function, it will be returned and we can check the error of WithActivation safely.